### PR TITLE
=act Fix ReceiveTimeoutTask was not cancelled bug when terminate an actor.

### PR DESCRIPTION
**Pull request by He-Pin**
References https://github.com/akka/akka/issues/31396

A regression introduced in https://github.com/akka/akka/pull/31059

@akilegaspi @johanandren This is ready for review.

### CODE REVIEW

**Comment by He-Pin** [comment]
My laptop has no power, will add tests tomorrow

**Comment by He-Pin** [comment]
@johanandren The test is still needed to cover this,will update when I soon.

File: akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala [hunk 1]
```diff
@@ -290,5 +302,14 @@ class ReceiveTimeoutSpec extends AkkaSpec() {
...
       probe.expectMsg(ReceiveTimeout)
     }
+
+    "Will cancel receive timeout task if stopped" in {
+      val probe = TestProbe()
+      val ref = system.actorOf(Props(new StoppingSelfActor(probe.ref)))
+      system.eventStream.subscribe(probe.ref, classOf[DeadLetter])
+      ref ! "Stop"
+      probe.expectMsg("stopping")
+      probe.expectNoMessage(3.seconds)
```
Comment (He-Pin) [inline akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala:312]: Added a test

Comment (johanandren) [reply akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala:312]: Great with a test, but 3 seconds that this test case will always block is a looooong time. 

Can the test set a much shorter receive timeout (say 200ms) immediately before stopping on receiving the stop message, and then wait double that here instead?

Comment (He-Pin) [reply akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala:312]: okay

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala [hunk 2]
```diff
@@ -290,5 +302,14 @@ class ReceiveTimeoutSpec extends AkkaSpec() {
...
     }
+
+    "Will cancel receive timeout task if stopped" in {
+      val probe = TestProbe()
+      system.eventStream.subscribe(probe.ref, classOf[DeadLetter])
+      val ref = system.actorOf(Props(new StoppingSelfActor(probe.ref)))
+      ref ! "Stop"
+      probe.expectMsg("stopping")
+      probe.expectNoMessage(200.millis * 2)
+    }
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala:313]: good

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by He-Pin** [comment]
@raboof @ennru Do you have any time to give an eye on this,thank you:)

**Pull request merged**